std/internal/math/biguintcore.d

```diff
@@ -1549,6 +1560,70 @@ char [] biguintToHex(char [] buff, const BigDigit [] data, char separator=0,
     return buff;
 }
 
+/**
+ * Convert a big uint into an octal string.
+ *
+ * Params:
+ *  buff = The destination buffer for the octal string. Must be large enough to
+ *      store the result, including leading zeroes, which is
+ *      ceil(data.length * BigDigitBits / 3) characters.
+ *      The buffer is filled from back to front, starting from `buff[$-1]`.
+ *  data = The biguint to be converted.
+ *
+ * Returns: The index of the leading non-zero digit in `buff`. Will be
+ * `buff.length` if the entire big uint is zero.
+ */
+size_t biguintToOctal(char[] buff, const(BigDigit)[] data)
+    pure nothrow @safe @nogc
+{
+    ubyte carry = 0;
+    int shift = 0;
+    size_t penPos = buff.length - 1;
+    size_t lastNonZero = buff.length;
```

Comment:
The code `writefln ("%o", BigInt (0));` now produces an empty string instead of a zero.
Perhaps `size_t lastNonZero = buff.length - 1;` would help to handle this corner case.
And please add `assert(format("%o", BigInt (0)) == "0");` to unittests.